src/makeCalculator.js

```diff
@@ -37,7 +37,51 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  const calculator = {
+    result: 0,
+
+    reset() {
+      calculator.result = 0;
+
+      return this;
+    },
+
+    operate(callback, number) {
+      callback(number);
+
+      return this;
+    },
+
+    add(num) {
+      calculator.result += num;
+
+      return this;
+    },
+
+    subtract(num) {
+      calculator.result -= num;
+
+      return this;
+    },
```

Comment:
make this functions like a [pure ones](https://www.geeksforgeeks.org/pure-functions-in-javascript/), pass your result as first param, and number as a second param. do it so your operations won't be dependant on previous result